ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/subscription/blockheaders/NewBlockHeadersSubscriptionServiceTest.java

```diff
@@ -63,76 +73,116 @@
   @Captor ArgumentCaptor<JsonRpcResult> responseCaptor;
 
   @Mock private SubscriptionManager subscriptionManager;
-  @Mock private BlockchainQueries blockchainQueries;
 
   private final BlockHeaderTestFixture blockHeaderTestFixture = new BlockHeaderTestFixture();
   private final TransactionTestFixture txTestFixture = new TransactionTestFixture();
   private final BlockHeader blockHeader = blockHeaderTestFixture.buildHeader();
   private final BlockResultFactory blockResultFactory = new BlockResultFactory();
+  private final BlockDataGenerator gen = new BlockDataGenerator();
+  private final BlockchainStorage blockchainStorage =
+      new KeyValueStoragePrefixedKeyBlockchainStorage(
+          new InMemoryKeyValueStorage(), new MainnetBlockHeaderFunctions());
+  private final Block genesisBlock = gen.genesisBlock();
+  private final DefaultBlockchain blockchain =
+      (DefaultBlockchain)
```

Comment:
What happens if this cast isn't here?

Reply:
I will change the variable to be a MutableBlockchain.